v2v-helper/pkg/utils/volumes_ext.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/platform9/vjailbreak/v2v-helper/vm"
+)
+
+// ToVolumeManageMap builds the request payload for manage volume.
+func ToVolumeManageMap(rdmDisk vm.RDMDisk) (map[string]interface{}, error) {
+	payload := map[string]interface{}{
+		"volume": map[string]interface{}{
+			"host": rdmDisk.CinderBackendPool,
+			"ref": map[string]string{
+				"source-name": fmt.Sprintf("volume-%s", rdmDisk.UUID),
+			},
+			"name":              rdmDisk.DiskName,
+			"volume_type":       rdmDisk.VolumeType,
+			"description":       rdmDisk.Description,
+			"bootable":          rdmDisk.Bootable,
+			"availability_zone": nil,
+		},
+	}
+	return payload, nil
+}
+
+// Manage triggers the volume manage request.
+
+func (osclient *OpenStackClients) CinderManage(rdmDisk vm.RDMDisk) (map[string]interface{}, error) {
+	body, err := ToVolumeManageMap(rdmDisk)
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
```

Comment:
The result of this operation has a "volume" key. I think it is OpenStack volume object. 
With that we can do it same as VMDisk object i.e. have `OpenstackVol` inside RDMDisk.

```go
type VMDisk struct {
	OpenstackVol    *volumes.Volume
}
```

Reply:
```
{
    "volume": {
        "host": null,
        "cluster": "cluster@backend",
        "ref": {
            "source-name": "existingLV",
            "source-id": "1234"
        },
        "name": "New Volume",
        "availability_zone": "az2",
        "description": "Volume imported from existingLV",
        "volume_type": null,
        "bootable": true,
        "metadata": {
            "key1": "value1",
            "key2": "value2"
        }
    }
}```

Response is volume object , we might have to do conversion , since i am only using id we can save ourselves from doing explicit conversion